graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/processor/ResponseMappingModel.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.processor;
+
+import graphql.Scalars;
+import graphql.language.FieldDefinition;
+import graphql.language.TypeName;
+import io.stargate.graphql.schema.schemafirst.util.TypeHelper;
+import java.util.EnumSet;
+import java.util.Optional;
+
+/**
+ * Represents a GraphQL object type that has been annotated with the {@code cql_payload} directive.
+ * This is a transient object (not mapped to a CQL table), that acts as a wrapper for operation
+ * responses.
+ */
+public class ResponseMappingModel {
+
+  private final Optional<EntityField> entityField;
+  private final EnumSet<TechnicalField> technicalFields;
+
+  public ResponseMappingModel(
+      Optional<EntityField> entityField, EnumSet<TechnicalField> technicalFields) {
+    this.entityField = entityField;
+    this.technicalFields = technicalFields;
+  }
+
+  /**
+   * The field in the response object that references a mapped entity (or list thereof). There is at
+   * most one such field.
+   */
+  public Optional<EntityField> getEntityField() {
+    return entityField;
+  }
+
+  /**
+   * An set of additional metadata fields that Stargate will know how to fill automatically. They
+   * have predefined names and types.
+   */
+  public EnumSet<TechnicalField> getTechnicalFields() {
+    return technicalFields;
+  }
+
+  public static class EntityField {
+
+    private final String name;
+    private final String entityName;
+    private final boolean isList;
+
+    public EntityField(String name, String entityName, boolean isList) {
+      this.name = name;
+      this.entityName = entityName;
+      this.isList = isList;
+    }
+
+    /** The name of the GraphQL field. */
+    public String getName() {
+      return name;
+    }
+
+    /** The entity that this field references. */
+    public String getEntityName() {
+      return entityName;
+    }
+
+    /** Whether the field is a list of entities, or a single entity. */
+    public boolean isList() {
+      return isList;
+    }
+  }
+
+  public enum TechnicalField {
+    APPLIED("applied", Scalars.GraphQLBoolean.getName()),
```

Comment:
just from curiosity, what other `TechnicalField`s are we anticipating?

Reply:
`PAGING_STATE` definitely (not relevant for inserts, but response payloads will be possible with selects too).

I thought there would be others, but looking at what is available on `ResultSet`, that might be it.